### PR TITLE
grayskull 2.3.0

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Requirements: https://github.com/conda/grayskull/blob/v2.3.0/pyproject.toml

Actions:
1. Add toml to host
2. Add semver >=2.13.0,<3.0.0 to run
3. Add host env for noarch generic subpackage (to address anaconda-linter error)
4. Add python to run for noarch generic subpackage (to address anaconda-linter error)
5. Skip 2 lints
6. Move test commands to the test scripts to resolve the `conda-build` or Prefect (?) bug

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
On win64 test failed without a proper error:

```
WARNING:conda_build.build:Tests failed for grayskull-2.3.0-py311haa95532_0.tar.bz2 - moving package to C:\b\abs_08fk4ksu_l\croot\broken

set PREFIX=C:\b\abs_08fk4ksu_l\croot\grayskull-split_1681217116605\_test_env
set SRC_DIR=C:\b\abs_08fk4ksu_l\croot\grayskull-split_1681217116605\test_tmp
%SRC_DIR%>call "%SRC_DIR%\conda_test_env_vars.bat"
%SRC_DIR%>set "CONDA_SHLVL="   &&
%SRC_DIR%>conda activate "%PREFIX%"
(%PREFIX%) %SRC_DIR%>IF 0 NEQ 0 exit /B 1
(%PREFIX%) %SRC_DIR%>"%PREFIX%\python.exe" -s "%SRC_DIR%\run_test.py"
import: 'grayskull'
import: 'grayskull'

(%PREFIX%) %SRC_DIR%>IF 0 NEQ 0 exit /B 1
(%PREFIX%) %SRC_DIR%>call "%SRC_DIR%\run_test.bat"
(%PREFIX%) %SRC_DIR%>pip check

No broken requirements found.

(%PREFIX%) %SRC_DIR%>IF 0 NEQ 0 exit /B 1
(%PREFIX%) %SRC_DIR%>grayskull --help

(%PREFIX%) %SRC_DIR%>IF 1 NEQ 0 exit /B 1
(%PREFIX%) %SRC_DIR%>IF 1 NEQ 0 exit /B 1
TESTS FAILED: grayskull-2.3.0-py311haa95532_0.tar.bz2
```

I'll try to move test commands to the test scripts

**Review by ELundby45 — COMMENTED** [review]
It doesn't seem like `run_test.bat` is being executed. 

I built this on a windows machine and inspected the generated package. Trying to execute grayskull.exe --help` errors out. I think you were getting actual failures on windows with the commands in the test section. 

```
grayskull-2.3.0-py310haa95532_0\Scripts\grayskull.exe
Fatal error in launcher: Unable to create process using '"C:\Users\dev-admin\Desktop\gray_ci\grayskull-split_1681234858731\_h_env\python.exe"  "C:\Users\dev-admin\Desktop\gray_ci\win-64\grayskull-2.3.0-py310haa95532_0\Scripts\grayskull.exe" ': The system cannot find the file specified.
```

It seems the exe is trying to use the build env's python?

**Comment by skupr-anaconda** [comment]
> It doesn't seem like `run_test.bat` is being executed.
> 
> I built this on a windows machine and inspected the generated package. Trying to execute grayskull.exe --help` errors out. I think you were getting actual failures on windows with the commands in the test section.
> 
> ```
> grayskull-2.3.0-py310haa95532_0\Scripts\grayskull.exe
> Fatal error in launcher: Unable to create process using '"C:\Users\dev-admin\Desktop\gray_ci\grayskull-split_1681234858731\_h_env\python.exe"  "C:\Users\dev-admin\Desktop\gray_ci\win-64\grayskull-2.3.0-py310haa95532_0\Scripts\grayskull.exe" ': The system cannot find the file specified.
> ```
> 
> It seems the exe is trying to use the build env's python?

@ELundby45 I wonder how to fix it...

**Comment by skupr-anaconda** [comment]
@ELundby45 I resolved this issue by moving `entry_points` to the output

**Comment by ELundby45** [comment]
> @ELundby45 I resolved this issue by moving `entry_points` to the output

Nice find!

**Review by ELundby45 — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,34 +1,45 @@
...
+    - setuptools_scm
+    - toml
+    - wheel
 
 outputs:
   - name: grayskull
     script: build_base.bat  # [win]
     script: build_base.sh   # [unix]
+    build:
+      skip: True  # [py<38]
```
Comment (danpetry) [inline recipe/meta.yaml]: is this required, since it's in the top-level package?

**Review by danpetry — APPROVED** [review]
minor nit, otherwise LGTM

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Removed

**Pull request merged**